Feature/impedance control

LuigiPenco93 opened the pull request:
* Extracted impedance control related things from reference-spreading branch
* added option to enable impedance control from RDX teleoperation panel and in behaviors in the hand pose action


Reference to impedance controller in TU/e paper:
Quadratic Programming-Based Reference Spreading
Control for Dual-Arm Robotic Manipulation With
Planned Simultaneous Impacts

LuigiPenco93 committed:
extracted from Coen's work, no reference spreading stuff

LuigiPenco93 committed:
messages for impedance

LuigiPenco93 committed:
fixed imports

LuigiPenco93 committed:
added flag for impedance in message hybrid traj

LuigiPenco93 committed:
added impedance option in behavior action

LuigiPenco93 commented on ihmc-high-level-behaviors/src/libgdx/java/us/ihmc/rdx/ui/affordances/RDXArmManager.java:
```diff
@@ -380,22 +380,15 @@ public void executeDesiredArmCommand(RobotSide robotSide)
             RDXBaseUI.pushNotification("Commanding arm jointspace trajectory...");
             communicationHelper.publishToController(armTrajectoryMessage);
          }
-         case TASKSPACE ->
```
removed taskspace without joints, as this is not the right way to control the arm and it's never worked.
As an external user, I want to either control the arm in jointspace or taskspace, with or without impedance control. The fact that the taskspace control is using a hybrid message shouldn't concern the user

LuigiPenco93 commented on ihmc-humanoid-robotics/src/main/java/us/ihmc/humanoidRobotics/communication/controllerAPI/command/HandHybridJointspaceTaskspaceTrajectoryCommand.java:
```diff
@@ -31,6 +44,41 @@ public HandHybridJointspaceTaskspaceTrajectoryCommand(RobotSide robotSide, boole
       this.taskspaceTrajectoryCommand.set(taskspaceTrajectoryCommand);
    }
 
+   public HandHybridJointspaceTaskspaceTrajectoryCommand(RobotSide robotSide, boolean forceExecution, SE3TrajectoryControllerCommand taskspaceTrajectoryCommand,
```
these messages are not used for now, but will when replaying back or using a planner that can provide the gain trajectories and feedforward trajectories

Beomyeong commented on ihmc-common-walking-control-modules/src/main/java/us/ihmc/commonWalkingControlModules/momentumBasedController/feedbackController/taskspace/SpatialFeedbackController.java:
```diff
@@ -328,15 +398,37 @@ public void submitFeedbackControlCommand(SpatialFeedbackControlCommand command)
+      dampingRatioMatrix.set(4,4, gains.getPositionGains().getDampingRatios()[1]);
+      dampingRatioMatrix.set(5,5, gains.getPositionGains().getDampingRatios()[2]);
       command.getSpatialAccelerationCommand().getSelectionMatrix(selectionMatrix);
+      selectionMatrix.getAngularPart().setAxisSelection(false, false, true);
```
Are the orientation for x and y axis always turned off?

Beomyeong commented on ihmc-common-walking-control-modules/src/test/java/us/ihmc/commonWalkingControlModules/momentumBasedController/feedbackController/taskspace/ImpedanceOrientationFeedbackControllerTest.java:
```diff
@@ -0,0 +1,140 @@
+         motionQPInputCalculator.convertSpatialAccelerationCommand(spatialControllerOutput, spatialMotionQPInput);
+
+         assertEquals(spatialMotionQPInput.taskJacobian, orientationMotionQPInput.taskJacobian, 1.0e-12);
+         assertEquals(spatialMotionQPInput.taskObjective, orientationMotionQPInput.taskObjective, 1.0e-12);
```
Was the failure of this test planned?

Beomyeong commented on ihmc-common-walking-control-modules/src/test/java/us/ihmc/commonWalkingControlModules/momentumBasedController/feedbackController/taskspace/ImpedancePointFeedbackControllerTest.java:
```diff
@@ -0,0 +1,446 @@
+
+      for (int i = 0; i < simulationTime / controlDT; i++)
+      {
+         pointFeedbackController.computeInverseDynamics();
```
It makes error, that jointIndices is null. 
Did you finish fix this? Or we should do ?

Beomyeong commented on ihmc-common-walking-control-modules/src/test/java/us/ihmc/commonWalkingControlModules/momentumBasedController/feedbackController/taskspace/ImpedanceSpatialFeedbackControllerTest.java:
```diff
@@ -0,0 +1,375 @@
+
+      for (int i = 0; i < simulationTime / controlDT; i++)
+      {
+         spatialFeedbackController.computeInverseDynamics();
```
It fails, 
ArrayIndexOutOfBoundsException.

stephenmcc commented on ihmc-common-walking-control-modules/src/main/java/us/ihmc/commonWalkingControlModules/momentumBasedController/feedbackController/taskspace/SpatialFeedbackController.java:
```diff
@@ -328,15 +396,37 @@ public void submitFeedbackControlCommand(SpatialFeedbackControlCommand command)
+      dampingRatioMatrix.set(4, 4, gains.getPositionGains().getDampingRatios()[1]);
+      dampingRatioMatrix.set(5, 5, gains.getPositionGains().getDampingRatios()[2]);
       command.getSpatialAccelerationCommand().getSelectionMatrix(selectionMatrix);
+      selectionMatrix.getAngularPart().setAxisSelection(false, false, true);
```
This should be removed

Beomyeong committed:
Fixing spatialFeedbackController.
Set the selectionMatrix.getAngularPar().setAxisSelection(false,false, true) makes error.

Beomyeong committed:
Fixing test fails

stephenmcc committed:
Fixing garbage creation in FB controllers. Appending message to naming

stephenmcc committed:
Removing impedance setter and getter from FB controller interface

LuigiPenco93 committed:
temp clean up

LuigiPenco93 committed:
removing PID3DGains and feedforward stuff for ref srpeading

LuigiPenco93 committed:
removing pid3gains stuff

LuigiPenco93 committed:
remvoing pid3gains stuff

LuigiPenco93 committed:
removing pid3dGains final

Commit:
undone selection matrix orientation

LuigiPenco93 committed:
1d orientation for testing

LuigiPenco93 committed:
removed gains trajectroies

LuigiPenco93 committed:
added impedance flag in KST output conveter

LuigiPenco93 committed:
added check box for vr

LuigiPenco93 committed:
kst parameters clean up

LuigiPenco93 committed:
Merge branch 'develop' into feature/impedance-control

LuigiPenco93 committed:
kst parameters passed as argument

stephenmcc committed:
Merge branch 'develop' into feature/impedance-control

stephenmcc committed:
Final cleanup pass: removing unused methods, consolidating hard-coded keyword, and applying formatter

Beomyeong committed:
Merge branch 'develop' into feature/impedance-control

Beomyeong committed:
Merge branch 'develop' into feature/impedance-control

Beomyeong committed:
Fixing NumericalInverseKinematicsCalculatorWithRobotTest

Beomyeong committed:
Merge remote-tracking branch 'origin/feature/impedance-control' into feature/impedance-control

Beomyeong committed:
Fixing load the fast_walking_parameters.xml file

Beomyeong committed:
Fixing FlatGroundQuickWalkingTest. Commented out the getWalkingControllerParameters.

Beomyeong committed:
Adding isImpedanceEnabled into equals method

Beomyeong committed:
Let disabled for the failing test.
The isImpedanceEnabled not changed randomly, and this fails tests.

LuigiPenco93 committed:
Merge branch 'develop' into feature/impedance-control

Beomyeong committed:
Merge remote-tracking branch 'origin/feature/impedance-control' into feature/impedance-control

Beomyeong committed:
Reducing the footsteps in generateFootstepForSteppingStonesB to pass the testSteppingStonesB

Beomyeong committed:
Merge branch 'develop' into feature/impedance-control

# Conflicts:
#	ihmc-avatar-interfaces/src/main/java/us/ihmc/avatar/networkProcessor/kinemtaticsStreamingToolboxModule/KinematicsStreamingToolboxParameters.java

Beomyeong committed:
Rollback the test steps

stephenmcc committed:
Fixing joint indexing

Beomyeong committed:
Fixed testConvergenceWithJerryQP
